### ISSUE TITLE
Plugin-Initialization is not thread save.

### ISSUE THREAD

**Issue by tom-englert** [VerifyTests/Verify#867]
Consider two test classes calling `if (!SomePlugin.Initialized) SomePlugin.Initialize();` - when those tests run in parallel, there is a chance that both call `Initialize()`, and either both run the initialization, or one test fails with an exception ("Already Initialized").

The "DotNet" recommended way to do lock-free and thread safe one time initialization is to do it in the static constructor, and leave the `Initialize()` method empty:

```c#
public class MyPluginClass {
    static MyPluginClass() 
    {
        VerifierSettings.RegisterFileConverter<TypeToDisassemble>(ConvertTypeDefinitionHandle);
        ...
    }

    public static bool Initialized { get; private set; }

    public static void Initialize()
    {
    }
```
Initialized can be safely ignored now, you can call now Initialize() as many time you like and from any thread.

**Comment by SimonCropp** [comment]
given the intended use for the Initialize api is to be called from the ModuleInitializer, do we really want to change this?

**Comment by tom-englert** [comment]
That will be fine if you're in the lucky position to be the sole developer on your own project, but with heterogeneous teams negotiating about and maintaining a single ModuleInitializer is a nightmare, so often Initialize is called from the static constructor of the test. Also ModuleInitalizer does not work out of the box in every framework, so it's a construct we tend to avoid, because it's too much hassle around it.

It would all work fine if a second call to Initialize is at least silently skipped, and does not throw an exception.

I could help to contribute the changes to some of the plugins, if you agree.

P.S. calling `VerifierSettings.InitializePlugins();` from the ModuleInitiailzer of `Verify` itself would make this a perfect solution, none would have to care about this any more - just reference you plugins and go!

**Comment by tom-englert** [comment]
Another issue with the ModuleInitializer is that you never can guarantee that there is only one ModuleInitializer, so even in this case Initializer might be called twice.

**Comment by SimonCropp** [comment]
sorry. not going to change the current behavior
